### PR TITLE
feat(queryTime): support doing calculations in a specific TZ [MA-1916]

### PR DESCRIPTION
**Pull request by adorack**
- Avoid breaking the API; `tz` is an optional param.
- Add many test cases.

# Summary

## PR Checklist

* [ ] **Naming & Structure:** the files and package structure use the conventions outlined in the [Creating a Package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md).
* [ ] **Tests pass:** check the output of all package unit and/or component tests.
  * [ ] If this PR is the result of a bug, test coverage was added accordingly.
* [ ] **Functional:** all changes do not break existing APIs, but if so, a BREAKING CHANGE commit is in place to bump the major version.
* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/public-ui-components#committing-changes).
* [ ] **Docs:** includes a technically accurate README, and the docs have been updated accordingly based on the changes in this PR.

### CODE REVIEW

**Comment by adorack** [comment]
Reverted to draft to figure out why some tests fail in CI.

**Review by mihai-peteu — CHANGES_REQUESTED** [review]
Looks good, nothing broken in konnect. Just a quick question on an unused param.

File: packages/analytics/analytics-utilities/src/timeframes.ts [hunk 1]
```diff
@@ -63,13 +69,13 @@ export class Timeframe {
 
   // rawEnd does not consider granularity and should not be used directly in queries.
   // Use `new QueryTime(timeframe, granularity?).queryEndSeconds()` instead.
-  rawEnd(): Date {
+  rawEnd(_tz?: string): Date {
```
[the remark this answers is not in the document]
Comment (adorack) [reply packages/analytics/analytics-utilities/src/timeframes.ts]: Done.

[the remark this answers is not in the document]
Comment (adorack) [reply packages/analytics/analytics-utilities/src/timeframes.ts:73]: Ah, never mind, I remember why I did this now -- the subclasses need it, and this is the base class, so I still need to declare it.  (If `TimeFrames` was implemented as an interface, I probably could leave it out, but I'd rather not change how the types are set up right now.)

Ignored the lint error instead.

**Review by mihai-peteu — APPROVED** [review]

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**